### ISSUE TITLE
Issue on creating multiple AMQP rhea clients

### ISSUE THREAD

**Issue by nmargaritis** [amqp/rhea#260]
Suppose that we have the following setup

- An APP using rhea that acts as a AMQP consumer (parses AMQP messages)
- A tests container (powered by mocha), part of the app, which runs the APP on test mode and creates a new rhea client to send messages to the APP. Essentially the test mode, actually produces some messages and tests the parsing output of the APP.

So based on this setup, we require to have 2 AMQP clients when the tests are running, essentially, we use the `.connect` method twice, since the APP source code is not related with the tests source code. However, since tests, start the app to test it, when the test suite runs, the app runs as well, together, on a `single node js process`.

The `single node process` along with the singleton implementation of the Rhea container (https://github.com/amqp/rhea/blob/master/dist/rhea.js#L9103)
```
module.exports = new Container();
```
results in the following error:
```
ConnectionError: Broker: localhost - Client: a4b85727-d0f8-ac43-a524-c4ecd51dda2a already connected from tcp://127.0.0.1:51010
```

Note that, module require is cached after the first time it is called. So since rhea exports a `new Contrainer`, if you have 2 files, on the same process, that require rhea, they will get the exact same instance (singleton). Or in other words, a container with the same UUID.

So my question is

> Is there a reason why, rhea library ships as a singleton? I know how to overcome the problem on my test suite, but my belief is that we should be able to create multiple clients if we wish to do it.

The library could either export the container class (no instance) or have a singleton wrapper that when `.connect` is called actually creates a new container and then calls `.connect` on that container.

A quick way to reproduce this error is this
```
var amqp1 = require('rhea');
let rhea1 = amqp1.connect(options);
var amqp2 = require('rhea');
let rhea2 = amqp2.connect(options);
```

**Comment by grs** [comment]
You can create as many containers as you like using the create_container() method. You can also override the container_id as an option to connect() and thus ensure different connections even from the same container have different container_ids if that is what you want.

**Comment by nmargaritis** [comment]
@grs Yes I have seen the create_container() method and the container id overwrite. I am using the latter on my tests suite setup.

I have created this issue as a question towards the singleton setup of rhea, the way it is now.

My feedback is that the lib could have a wrapper class that is actually the singleton (rhea), cause currently it returns an instance of a container, and that instance allows you to create multiple containers, if you don't do it, you still have the same instance at hand. Also it requires you to do such manipulation, while it could work with the constructor approach (`new Object()`).

The current setup is a little bit confusing, logically.

**Comment by grs** [comment]
My view was that the default behaviour is the simplest for the simple cases (i.e. multiple connections would share a library generated container id indicating they originated from the same process), and that other cases could be easily handled.

**Comment by martppa** [comment]
Hi, I don't know if you actually want on purpose to have multiple connections from the same client, but in my case for example, I own serveral packages thats connects to an ActiveMQ server bound to an app  and what I use to avoid the issue is to set rhea as peerDependency, keep only one open connection and reuse it among all packages.

**Comment by sunil-rnl** [comment]
@grs When I'm creating multiple connection using create_container() method. I'm getting this error

```
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 accepted listeners added to [Container]. Use emitter.setMaxListeners() to increase limit
```

**Comment by grs** [comment]
@sunil-rnl that sounds like you are adding a listener many times. The create_container() method itself does not do setup any listeners so I suspect it might be your application code. Do you have a reproducer? (This is also really a different issue)

**Comment by sunil-rnl** [comment]
@grs  I have opened a new issue https://github.com/amqp/rhea/issues/363
